array.c

```diff
@@ -2991,22 +2991,46 @@ rb_ary_join(VALUE ary, VALUE sep)
 
 /*
  *  call-seq:
- *     ary.join(separator=$,)    -> str
+ *    array.join ->new_string
+ *    array.join(separator = nil) -> new_string
```

Comment:
The `nil` here should still be `$,`, I think:

```
$ ruby -e '$, = "x";p %w[1 2].join' 
"1x2"
```

Reply:
Thanks.  So true.  Will fix.